src/data/fin.lean

```diff
@@ -685,6 +685,24 @@ lemma forall_fin_succ {P : fin (n+1) → Prop} :
   (∀ i, P i) ↔ P 0 ∧ (∀ i:fin n, P i.succ) :=
 ⟨λ H, ⟨H 0, λ i, H _⟩, λ ⟨H0, H1⟩ i, fin.cases H0 H1 i⟩
 
+
```

Comment:
This is a generalization of `fin.cases` above, so I recommend moving this above `fin.cases`, changing the order of arguments to make them consistent, and then replacing the proof of `fin.cases` with a one-liner.

Reply:
I added `fin.induction` on its own as well, which is the argument order in `fin.cases`.